layers/+chat/circe/packages.el

```diff
@@ -0,0 +1,162 @@
+;;
+;; Copyright (c) 2012-2017 Sylvain Benner & Contributors
+;;
+;; Author: Chris Barrett <chris.d.barrett@me.com>
+;; Author: Aidan Nyquist <contact@aidannyquist.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Code:
+
+(defconst circe-packages
+  '(circe
+    persp-mode
+    circe-notifications
+    helm
+    helm-circe
+    ))
+
+(defun circe/init-circe ()
+  (use-package circe
+    :commands circe
+    :config
+    (progn
+      (set-face-background 'circe-prompt-face nil)
+      (setq circe-reduce-lurker-spam t)
+      (setq circe-use-cycle-completion t)
+      (setq circe-active-users-timeout (* 60 30)) ; 30 minutes
+      (setq circe-format-say "{nick}> {body}")
+      (setq circe-format-self-say (concat (propertize ">>>" 'face 'circe-self-say-face) " {body}"))
+      (setq circe-prompt-string (concat (propertize ">>>" 'face 'circe-prompt-face) " "))
+      (setq circe-highlight-nick-type 'all)
+
+      ;; Keybindings
+      (spacemacs/set-leader-keys "aic" 'circe)
+
+      (dolist (mode circe-modes)
+        (progn
+        (spacemacs/declare-prefix-for-mode mode "mc" "command")
+        (spacemacs/set-leader-keys-for-major-mode mode
+          "r" 'circe-reconnect
+          "R" 'circe-reconnect-all
+          "v" 'circe-version
+          "c n" 'circe-command-NAMES
+          "c q" 'circe-command-QUERY
+          "c h" 'circe-command-HELP
+          "c t" 'circe-command-CHTOPIC 
+          "c s" 'circe-command-SAY
+          )))
+
+      ;; Show channel name in prompt.
+
+      (defun circe/bufname-to-channame (s)
+        (-let* (((_ gitter-fmt) (s-match (rx (+ nonl) "/" (group (+ nonl)) eos) s))
+                ((_ standard-fmt) (s-match (rx "#" (group (+ nonl)) eos) s)))
+          (or gitter-fmt standard-fmt)))
+
+      (defun circe/set-prompt ()
+        (let* ((chan (circe/bufname-to-channame (buffer-name)))
+               (prompt (concat "#" (propertize chan 'face 'circe-prompt-face) " > ")))
+          (lui-set-prompt prompt)))
+
+      (add-hook 'circe-chat-mode-hook #'circe/set-prompt)
+
+      ;; Timestamps in margins.
+
+      (setq lui-time-stamp-position 'right-margin)
+      (setq lui-time-stamp-format "%H:%M")
+      (setq lui-fill-type nil)))
+
+  (use-package circe-color-nicks
+    :commands enable-circe-color-nicks
+    :init
+    (with-eval-after-load 'circe
+      (enable-circe-color-nicks))
+    :config
+    (setq circe-color-nicks-everywhere t))
+
+  (use-package lui
+    :init
+    (progn
+      (defun circe/set-local-vars ()
+        (setq fringes-outside-margins t)
+        (setq right-margin-width 5)
+        (setq word-wrap t)
+        (setq wrap-prefix "    "))
+
+      (add-hook 'lui-mode-hook #'circe/set-local-vars)))
+
+  (use-package lui-autopaste
+    :commands enable-lui-autopaste
+    :init
+    (add-hook 'circe-channel-mode-hook #'enable-lui-autopaste)))
+
+(defun circe/post-init-helm ()
+  (use-package helm
```

Comment:
the only place a `use-package` declaration should be located is the `init` function. Otherwise you are calling it twice.